internal/mode/static/telemerty.go

```diff
@@ -0,0 +1,129 @@
+package static
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type errorHandler struct {
+	logger logr.Logger
+}
+
+func (h errorHandler) Handle(err error) {
+	h.logger.Error(err, "error in telemetry")
+}
+
+type telemetryReporter struct {
+	k8sClient client.Client
+	logger    logr.Logger
+}
+
+func (r *telemetryReporter) Start(ctx context.Context) error {
+	// runs once in a POC.
+
+	// telemetry endpoint configuration
+	var (
+		// /otel-collector/README.md deploys a collector with the endpoint below
+		// no TLS or auth headers are needed for that collector
+		endpoint = "my-opentelemetry-collector.default.svc.cluster.local:4317"
+		secure   = false               // use TLS or not.
+		headers  = map[string]string{} // allows t add headers to a GRPC connection. (e.g. authentication)
+	)
+
+	// gather telemetry data
+	var ns v1.Namespace
+	err := r.k8sClient.Get(ctx, client.ObjectKey{Name: "kube-system"}, &ns)
+	if err != nil {
+		return err
+	}
+
+	// cluster ID (UUID of kube-system namespace)
+	clusterID := string(ns.UID)
+	// NGF version
+	ngfVersion := "product-tel-prototype-0.0.1"
+
+	// configure OTel
+	otel.SetLogger(r.logger)
```

Comment:
What portion of this function is initialization? Lines 58-91? Trying to figure out what this will look like when it runs continuously.

Reply:
@kate-osborn 

I added some additional comments in the last commit. Also, I added Steps comments in the code, which are used below.

I see two approaches:
(1)
Step 1 - once
Steps 2-7 - every time (assuming period of ~1day) we do reporting

(2)
Steps 1-5 - once
Steps 6-7 - every time we do reporting 

Note that running this code https://github.com/pleshakov/nginx-kubernetes-gateway/commit/bd0d95d6f4790e8acf915f217c8f34678c3cf0d2 , after the first span is sent, the the corresponding network connection between NGF and Collector stays open (confirmed by netstat) (at least after 20 minutes after sending previous span).  I don't think we should keep the connection open if we send data infrequently

Reply:
What's the benefit of approach #1? 

If the network connection isn't established until we start sending data, it seems to me like the best option is approach #2.

Reply:
if the data is sent not frequently, then maintaining a persistent connection is an overkill, especially on the backend side.
with the shared code, the connection is not established until we start sending data, but then it kept open.

Reply:
> the connection is not established until we start sending data, but then it kept open.

This is the piece I was missing. Makes sense 👍